src/server.js

```diff
@@ -1,168 +1,161 @@
-        }),
-    ],
-});
 
-const app = koa();
-const pool = createPhantomPool({
-    min: process.env.SCREENIE_POOL_MIN || 2,
-    max: process.env.SCREENIE_POOL_MAX || 10,
+const logger = new winston.Logger({
+  level: process.env.SCREENIE_LOG_LEVEL || 'info',
+  transports: [
+    new winston.transports.Console({
+      timestamp: () => new Date().toISOString()
+    })
+  ]
 });
+
+const app = new Koa();
 const serverPort = process.env.SCREENIE_PORT || 3000;
 
 const imageSize = {
-    width: process.env.SCREENIE_WIDTH || 1024,
-    height: process.env.SCREENIE_HEIGHT || 768,
+  width: process.env.SCREENIE_WIDTH || 1024,
+  height: process.env.SCREENIE_HEIGHT || 768
 };
-const supportedFormats = [
-    'gif',
-    'jpeg',
-    'jpg',
-    'pdf',
-    'png',
-];
+const supportedFormats = ['jpg', 'jpeg', 'pdf', 'png'];
 const defaultFormat = process.env.SCREENIE_DEFAULT_FORMAT || 'jpeg';
+const chromiumArgs = process.env.SCREENIE_CHROMIUM_ARGS
+  ? { args: process.env.SCREENIE_CHROMIUM_ARGS.split(' ') }
+  : {};
+const chromiumExec = process.env.SCREENIE_CHROMIUM_EXEC
+  ? { executablePath: process.env.SCREENIE_CHROMIUM_EXEC }
+  : {};
 
-/*
- * Clean up the PhantomJS pool before exiting when receiving a termination
- * signal. Exit with status code 143 (128 + SIGTERM's signal number, 15).
- */
-process.on('SIGTERM', () => {
-    logger.info('Received SIGTERM, exiting...');
-    pool.drain()
-        .then(() => pool.clear())
-        .then(() => process.exit(143));
-});
+let browser;
 
 /**
- * Set up a PhantomJS instance with a page and configure viewport size.
+ * Set up a Puppeteer instance with a page and configure viewport size.
  */
-app.use(function *(next) {
-    const size = {
-        width: Math.min(
-            2048,
-            parseInt(this.request.query.width, 10) || imageSize.width
-        ),
-        height: Math.min(
-            2048,
-            parseInt(this.request.query.height, 10) || imageSize.height
-        ),
-    };
-
-    logger.verbose(
-        `Instantiating PhantomJS with page size ${size.width}x${size.height}`
-    );
-
-    yield pool.use(instance => instance.createPage())
-        .then(page => this.state.page = page)
-        .then(() => this.state.page.property('viewportSize', size));
-
-    yield next;
+app.use(function*(next) {
+  const { width, height } = this.request.query;
+  const size = {
+    width: Math.min(2048, parseInt(width, 10) || imageSize.width),
+    height: Math.min(2048, parseInt(height, 10) || imageSize.height)
+  };
+  let error;
+
+  logger.verbose(
+    `Instantiating Puppeteer with page size ${size.width}x${size.height}`
+  );
+
+  if (!browser) {
+    yield puppeteer
+      .launch({ ...chromiumExec, ...chromiumArgs })
+      .then(instance => (browser = instance));
+  }
+
+  yield browser
+    .newPage()
+    .then(page => (this.state.page = page))
+    .then(() => this.state.page.setViewport(size))
+    .catch(error => (error = true));
```

Comment:
The `error` argument to the arrow function here will overshadow the `let error` definition above, so this can't work as intended.

Reply:
Also related to error handling, maybe we should try fix this warning from Koa?

> http-errors deprecated non-first-argument status code; replace with createError(400, ...) node_modules/koa/lib/context.js:89:23